core/eolearn/core/visualizations.py

```diff
@@ -0,0 +1,339 @@
+        """
+        crs = self.eopatch.bbox.crs
+        crs = CRS.POP_WEB if crs == CRS.WGS84 else crs
+        data_da = array_to_dataframe(self.eopatch, (FeatureType.DATA, feature_name), crs=crs)
+        if self.mask:
+            data_da = self.mask_data(data_da)
+        timestamps = self.eopatch.timestamp
+        crs = self.eopatch.bbox.crs
+        if not self.rgb:
+            return data_da.hvplot(x='x', y='y', crs=ccrs.epsg(int(crs.value)))
+        data_rgb = self.eopatch_da_to_rgb(data_da, feature_name, crs)
+        rgb_dict = {timestamp_: self.plot_rgb_one(data_rgb, timestamp_) for timestamp_ in timestamps}
+
+        return hv.HoloMap(rgb_dict, kdims=['time'])
+
+    @staticmethod
+    def plot_rgb_one(eopatch_da, timestamp):  # OK
+        """ Returns visualization for one timestamp for FeatureType.DATA
+        :param eopatch_da: eopatch converted to xarray DataArray
+        :type eopatch_da: xarray DataArray
+        :param timestamp: timestamp to make plot for
+        :type timestamp: datetime
+        :return: visualization
+        :rtype:  holoviews/geoviews/bokeh
+        """
+        return eopatch_da.sel(time=timestamp).drop('time').hvplot(x='x', y='y')
+
+    def plot_raster(self, feature_type, feature_name):
+        """ Makes visualization for raster data (except for FeatureType.DATA)
+
+        :param feature_type: type of eopatch feature
+        :type feature_type: FeatureType
+        :param feature_name: name of eopatch feature
+        :type feature_name: str
+        :return: visualization
+        :rtype: holoviews/geoviews/bokeh
+        """
+        crs = self.eopatch.bbox.crs
+        crs = CRS.POP_WEB if crs == CRS.WGS84 else crs
+        data_da = array_to_dataframe(self.eopatch, (feature_type, feature_name), crs=crs)
+        data_min = data_da.values.min()
+        data_max = data_da.values.max()
+        data_levels = len(np.unique(data_da))
+        data_levels = 11 if data_levels > 11 else data_levels
+        data_da = data_da.where(data_da > 0).fillna(-1)
+        vis = data_da.hvplot(x='x', y='y',
+                             crs=ccrs.epsg(int(crs.value))).opts(clim=(data_min, data_max),
+                                                                 clipping_colors={'min': 'transparent'},
+                                                                 color_levels=data_levels)
+        return vis
+
+    def plot_vector(self, feature_name):
+        """ Visualizaton for vector (FeatureType.VECTOR) data
+
+        :param feature_name: name of eopatch feature
+        :type feature_name: str
+        :return: visualization
+        :rtype: holoviews/geoviews/bokeh
+
+        """
+        crs = self.eopatch.bbox.crs
+        timestamps = self.eopatch.timestamp
+        data_gpd = self.fill_vector(FeatureType.VECTOR, feature_name)
+        if crs == CRS.WGS84:
+            crs = CRS.POP_WEB
+            data_gpd = data_gpd.to_crs({'init': 'epsg:'+crs.value})
```

Comment:
A minor style recommendation: instead of `'epsg:'+crs.value` I think it is nicer to use `epsg:{}`.format(crs.value).

Besides the visual affect, there is also 1 important advantage of using `format` function over string summation. Parameters of `format` function will be automatically cast to `str` and you don't have to worry about that. However with string summation you always have to be careful not to sum together e.g. a sting and an integer.
In this case `crs.value` is a string but maybe in the future versions of sentinelhub-py this can change. So if you use format here, you don't have to worry about that.

Reply:
changed for all `to_crs()` functions